File: jdbc-shadow/src/test/java/com/gabry/shadow/kyuubi/TestUtils.java
package com.gabry.shadow.kyuubi;

import java.sql.ResultSet;
import java.sql.ResultSetMetaData;
import java.sql.SQLException;

public class TestUtils {
  private TestUtils() {
    // do nothing
  }

  public static void printResultSet(ResultSet resultSet) throws SQLException {
    ResultSetMetaData metaData = resultSet.getMetaData();
    for (int i = 1; i <= metaData.getColumnCount() - 1; i++) {
      System.out.print(metaData.getColumnName(i) + "(" + metaData.getColumnTypeName(i) + ")\t");
    }
    System.out.println(
        metaData.getColumnName(metaData.getColumnCount())
            + "("
            + metaData.getColumnTypeName(metaData.getColumnCount())
            + ")");
    while (resultSet.next()) {
      for (int i = 1; i <= metaData.getColumnCount() - 1; i++) {
        System.out.print(resultSet.getObject(i) + "\t");
      }
      System.out.println(resultSet.getObject(metaData.getColumnCount()));
    }
  }
}

File: unit-tests/src/test/scala/com/gabry/kyuubi/test/jdbc/TestUtils.scala
package com.gabry.kyuubi.test.jdbc

import org.slf4j.LoggerFactory

import java.sql.ResultSet

object TestUtils {
  private val logger = LoggerFactory.getLogger(TestUtils.getClass)

  def printResult(resultSet: ResultSet): Unit = {
    val metadata = resultSet.getMetaData
    val columnTypes = 1 to metadata.getColumnCount map (i => s"${metadata.getColumnName(i)}(${metadata.getColumnTypeName(i)})")
    println(columnTypes.mkString("\t"))
    while (resultSet.next()) {
      val columnValues = 1 to metadata.getColumnCount map (i => resultSet.getObject(i))
      println(columnValues.mkString("\t"))
    }
  }
}
